capstone/src/main/webapp/data_viz.js

```diff
@@ -168,4 +153,132 @@ function mouseInToRegion(e) {
+        if(parsed_data.length === 10){
+          let date = parsed_data[0];
+          let county = parsed_data[1];
+          let state = parsed_data[2];
+          let fips = parsed_data[3];
+          let cases = parsed_data[4];
+          let deaths = parsed_data[5];
+          let confirmed_cases = parsed_data[6];
+          let confirmed_deaths = parsed_data[7];
+          let probable_cases = parsed_data[8];
+          let probable_deaths = parsed_data[9];
+          let key = county + '-' + fips;
+          //note that there may be blank '' attributes
+          let county_data = {"date": date, "state": state, "fips": fips, "cases": cases,
+            "deaths": deaths, "confirmed_cases": confirmed_cases, "confirmed_deaths": confirmed_deaths, 
+            "probable_cases": probable_cases, "probable_deaths": probable_deaths};
+          nyt_data[key] = county_data;
+          county_data["county"] = county;
+          county_data["fips"] = fips;
+          if(nyt_data_by_state[state] !== undefined){
+            nyt_data_by_state[state].push(county_data);  
+          }else{
+            state_names.push(state);
+            nyt_data_by_state[state] = [county_data];
+          }
+        }
+      }
+    }
+
+    let state_counts = {};
+    if(variable === "cases"){
+      state_counts = setStateWideCases(state_names, nyt_data_by_state);
+    }
+    if(variable === "deaths"){
+      state_counts = setStateWideDeaths(state_names, nyt_data_by_state);
+    }
+    setData(state_counts);
+    let range = findMinMax(state_names, state_counts);
+    localStorage.setItem("min", range.min);
+    localStorage.setItem("max", range.max);
+    document.getElementById('data-min').textContent = range.min.toLocaleString();
+    document.getElementById('data-max').textContent = range.max.toLocaleString();
+  });
+}
+
+/* finds the min and max case or death counts for the legend */
+function findMinMax(state_names, state_counts){
+  let min = Number.MAX_VALUE;
+  let max = -Number.MAX_VALUE;
+  for(let i = 0; i < state_names.length; i++){
+    if(STATES_DISPLAYED.indexOf(state_names[i]) > -1){
+      let count = state_counts[state_names[i]];
+      if(min > count){
+        min = count;
+      }
+      if(max < count){
+        max = count;
+      }
+    }
+  }
+  return {"min": min, "max": max};
+}
+
+/* load data for graph */
+function setData(state_counts){
+  map.data.forEach(function(row) {
+    row.setProperty('input_variable', state_counts[row.j.NAME]);
+  });
+}
+
+/* sum the data for the each of the counties in the state to estimate statewide death count */
+function totalDeathsState(state, counties){
+  let death_count = 0;
+  for(let i = 0; i < counties.length; i++){
+    let deaths = parseInt(counties[i].deaths);
+    if(deaths !== NaN){
+      death_count += deaths;
+    }
+  }
+  return death_count;
+}
+
+/* sum the data for the each of the counties in the state to estimate statewide case count */
+function totalCasesState(state, counties){
```

Comment:
Be very clear when using the name "state". It can be confused with the application state.